File: platform/polyglot/figma_api/mjs/src/routes/webhooks.mjs
/**
 * Webhooks Routes — Figma API SDK (v2 API)
 */

export default async function webhooksRoutes(server, { webhooksClient }) {
  server.get('/webhooks/:webhookId', async (request) => {
    return webhooksClient.getWebhook(request.params.webhookId);
  });

  server.get('/teams/:teamId/webhooks', async (request) => {
    return webhooksClient.listTeamWebhooks(request.params.teamId);
  });

  server.post('/webhooks', async (request, reply) => {
    const result = await webhooksClient.createWebhook(request.body.team_id, request.body);
    return reply.status(201).send(result);
  });

  server.put('/webhooks/:webhookId', async (request) => {
    return webhooksClient.updateWebhook(request.params.webhookId, request.body);
  });

  server.delete('/webhooks/:webhookId', async (request) => {
    return webhooksClient.deleteWebhook(request.params.webhookId);
  });

  server.get('/webhooks/:webhookId/requests', async (request) => {
    return webhooksClient.getWebhookRequests(request.params.webhookId);
  });

  return Promise.resolve();
}

File: platform/polyglot/figma_api/py/figma_api/routes/webhooks.py
"""Webhooks Routes — Figma API SDK (v2 API)"""

from typing import Any, Dict, Optional

from fastapi import APIRouter, Depends, Request

router = APIRouter()


def get_webhooks_client(request: Request):
    return request.app.state.webhooks_client


@router.get("/webhooks/{webhook_id}")
async def get_webhook(webhook_id: str, client=Depends(get_webhooks_client)):
    return await client.get_webhook(webhook_id)


@router.get("/teams/{team_id}/webhooks")
async def list_team_webhooks(team_id: str, client=Depends(get_webhooks_client)):
    return await client.list_team_webhooks(team_id)


@router.post("/webhooks", status_code=201)
async def create_webhook(body: Dict[str, Any], client=Depends(get_webhooks_client)):
    return await client.create_webhook(
        body["team_id"],
        event_type=body["event_type"],
        endpoint=body["endpoint"],
        passcode=body.get("passcode"),
        status=body.get("status"),
        description=body.get("description"),
    )


@router.put("/webhooks/{webhook_id}")
async def update_webhook(
    webhook_id: str, body: Dict[str, Any], client=Depends(get_webhooks_client)
):
    return await client.update_webhook(webhook_id, body)


@router.delete("/webhooks/{webhook_id}")
async def delete_webhook(webhook_id: str, client=Depends(get_webhooks_client)):
    return await client.delete_webhook(webhook_id)


@router.get("/webhooks/{webhook_id}/requests")
async def get_webhook_requests(webhook_id: str, client=Depends(get_webhooks_client)):
    return await client.get_webhook_requests(webhook_id)
